include/jwt-cpp/jwt.h

```diff
@@ -396,6 +396,60 @@ namespace jwt {
 			return res;
 		}
 
+		/**
+		 * \brief Convert the certificate provided as base64 DER to PEM.
+		 *
+		 * This is useful when using with JWKs as x5c claim is encoded as base64 DER. More info
+		 * (here)[https://tools.ietf.org/html/rfc7517#section-4.7]
+		 *
+		 * \param cert_base64_der_str String containing the certificate encoded as base64 DER
+		 * \param ec		error_code for error_detection (gets cleared if no error occures)
+		 */
+		inline
+		std::string convert_base64_der_to_pem(const std::string& cert_base64_der_str, std::error_code& ec) {
+			ec.clear();
+			auto decodedStr = base::decode<alphabet::base64>(base::pad<alphabet::base64>(cert_base64_der_str));
+			auto c_str = reinterpret_cast<const unsigned char *> (decodedStr.c_str());
+			//
+			// d2i -> DER to internal x509 struct
+			//
+			std::unique_ptr<X509, decltype(&X509_free)> cert(d2i_X509(NULL, & c_str, decodedStr.size()), X509_free);
```

Comment:
I also checked, https://github.com/openssl/openssl/blame/e84b663a933eff8834d66233ac0d6882ed918b82/doc/crypto/d2i_X509.pod https://www.openssl.org/docs/man1.0.2/man3/d2i_X509.html

Should be safe it's in the 1.0.X

```suggestion
			std::unique_ptr<X509, decltype(&X509_free)> cert(d2i_X509(NULL, & c_str, decodedStr.size()), X509_free);
```